### PR TITLE
Log when update AffixSetting using addAffixMapUpdateConsumer

### PR DESCRIPTION
**Pull request by Rassyan**
When updateing settings, most of SettingUpdater will call `org.elasticsearch.common.settings.Setting#logSettingUpdate` to log the difference between before and after. However, since version 6.1.0, the new subclass `AffixSetting` will not log any message when using `org.elasticsearch.common.settings.Setting.AffixSetting#newAffixMapUpdater` method to update settings. Currently, the scope of influence is this three settings in `FilterAllocationDecider`: `cluster.routing.allocation.require`, `cluster.routing.allocation.include`,`cluster.routing.allocation.exclude`, which are not easy to notice. Call `logSettingUpdate` in the same way as `GroupSetting`, I think it's ok to fix this issue. 

Manual test case after this fix:
```
PUT _cluster/settings
{
  "transient": {
    "cluster.routing.allocation.exclude._ip": "1.1.1.1",
    "cluster.routing.allocation.exclude._name": "abcde"
  }
}
```

`[2023-06-25T15:27:05,672][INFO ][o.e.c.s.ClusterSettings  ] [node1] updating [cluster.routing.allocation.exclude.] from [{}] to [{"cluster.routing.allocation.exclude._ip":"1.1.1.1","cluster.routing.allocation.exclude._name":"abcde"}]`

```
PUT _cluster/settings
{
  "transient": {
    "cluster.routing.allocation.exclude._ip": "1.1.1.1",
    "cluster.routing.allocation.exclude._name": null
  }
}
```

`[2023-06-25T15:28:07,868][INFO ][o.e.c.s.ClusterSettings  ] [node2] updating [cluster.routing.allocation.exclude.] from [{"cluster.routing.allocation.exclude._ip":"1.1.1.1","cluster.routing.allocation.exclude._name":"abcde"}] to [{"cluster.routing.allocation.exclude._ip":"1.1.1.1"}]`

### CODE REVIEW

**Comment by Rassyan** [comment]
@elasticsearchmachine Could you please assist with a code review for this pull request? It has been open for a while, and I would greatly appreciate your feedback. Thank you!

**Comment by mosche** [comment]
@elasticmachine test this please

**Comment by mosche** [comment]
@elasticmachine update branch

**Comment by mosche** [comment]
Thank you so much for your contribution @Rassyan. I am really sorry it took us so long to respond, my apologies!

I had some discussions around this with my colleagues. We're a bit reluctant to support `innerGetRaw` for any groups of settings, a raw setting value simply doesn't make much sense in these cases (note, `getRaw` calls `innerGetRaw`). Unfortunately that is already the case for `GroupSetting`, that part of the code is fairly old and turns out to be a poor example.

Having that said, I suggest adding a new package private method `getLogString(Settings settings)` or similar that calls `getRaw(Settings settings)` for use in `logSettingUpdate` (rather than `getRaw` itself). `getLogString` could then be safely overwritten in `AffixSetting` without any further impact on other parts of the code.

Please let me know in case you have questions or if this doesn't make sense to you.

**Comment by Rassyan** [comment]
Dear @mosche ,

I hope this message finds you well. First and foremost, I would like to extend my sincere apologies for the delayed response to your valuable feedback. I appreciate the time and effort you've invested in reviewing my initial submission.

Following your suggestions, I have implemented a new package-private method `getLogString(Settings settings)` within the `Setting` class. This method is designed to encapsulate the logic for generating a log-friendly string representation of the settings, which is now invoked by the `logSettingUpdate` method.

Additionally, I have overridden the `getLogString` method in the `AffixSetting` class to ensure that the log messages for affix settings are both informative and formatted correctly. This approach aligns with your guidance and maintains the integrity of the existing codebase while providing a pathway for future enhancements, particularly concerning the `GroupSetting` class.

I have refrained from making any changes to the `GroupSetting` class at this time, as I understand that it serves as a legacy example. However, with the new `getLogString` method in place, there is now an opportunity to refactor `GroupSetting` in the future, should it be deemed necessary.

The commit has been force-pushed, and I kindly request your review of the changes at your earliest convenience. I am eager to hear your thoughts and am open to further discussion or revisions as needed.

Thank you once again for your guidance and patience. I am looking forward to your feedback.

Warm regards,

Rassyan

**Comment by mosche** [comment]
Thanks a lot for your changes @Rassyan.
This is looking good, just waiting for tests to pass 👍

File: server/src/main/java/org/elasticsearch/common/settings/Setting.java [hunk 1]
```diff
@@ -1008,6 +1016,20 @@ public String innerGetRaw(final Settings settings) {
             );
         }
 
+        @Override
+        public String getLogString(final Settings settings) {
```
Comment (mosche) [inline server/src/main/java/org/elasticsearch/common/settings/Setting.java:1020]: Could you fix visibility here, pls.

**Review by mosche — COMMENTED** [review]
@Rassyan Thanks so much. LGTM, except for a small point. 
I'll merge once fixed, please have a look below.

File: server/src/main/java/org/elasticsearch/common/settings/Setting.java (hunk 1 above)
Comment (Rassyan) [reply server/src/main/java/org/elasticsearch/common/settings/Setting.java]: Sure, it's done.

**Review by mosche — APPROVED** [review]

**Pull request merged**